libraries/AP_HAL/DSP.cpp

```diff
@@ -64,57 +68,40 @@ DSP::FFTWindowState::~FFTWindowState()
 {
     hal.util->free_type(_freq_bins, sizeof(float) * (_window_size), DSP_MEM_REGION);
```

Comment:
under what circumstances does this destructor run?

Reply:
It is run if any of the allocations fail (the object is deleted in fft_init())